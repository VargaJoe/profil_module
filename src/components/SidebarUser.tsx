import * as React							from 'react';
import { connect }             				from 'react-redux';
import { Link } 							from 'react-router-dom';

const DATA = require('../config.json');

interface Props {
	userName: string;
	user: any;
	fullName: string;
}

export class SidebarUserComponent extends React.Component<Props, any> {
	constructor(props: any) {
        super(props);
        this.state = {
            usName : this.props.fullName,
            usAvatar: '',
		};
		
		this.isEmpty = this.isEmpty.bind(this);
	}
	
	isEmpty = (obj: object) => {
		for (var key in obj) {
			if ( obj.hasOwnProperty(key) ) {
				return false;
			}
		}
		return true;
	}
	
	render () {
		let imageSrc = !this.props.user.AvatarImageRef ? 'empty' : this.props.user.AvatarImageRef.Path ;
			return (
				<Link to={'/user/' + this.props.userName} >
					<div className="sn_sidebar__user">
					{ this.isEmpty(this.props.user) ? 
						(
							<span className="sn_sidebar__user__avatar--default">
								<i className="fi flaticon-user" />
							</span>
						)
					: 
						(
							<img src={DATA.domain + imageSrc} alt={this.props.user.FullName} className="sn_sidebar__user__avatar"/>
						)
					}
						<div className="sn_sidebar__user__name">
							{this.isEmpty(this.props.user) ? this.state.usName : this.props.user.FullName}
						</div>
					</div>
				</Link>
			);
		}
}

const mapStateToProps = (state: any) => {
	return {
		userName: 			state.sensenet.session.user.userName, // state.user.user.FullName,
		fullName: 			state.sensenet.session.user.fullName, 
		user: 				state.user.user
	};
};

export default connect(
	mapStateToProps
)(SidebarUserComponent as any);
