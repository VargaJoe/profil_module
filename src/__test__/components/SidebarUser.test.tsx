jest.unmock('../../index.tsx');
jest.unmock('redux-mock-store');

import * as React 							from 'react';
import SidebarUser 							from '../../components/SidebarUser';
import { 
	configure, 
    mount } 								from 'enzyme';
import * as Adapter 						from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';
import * as createMockStore 				from 'redux-mock-store';
import { Provider } 						from 'react-redux';
import {
    BrowserRouter as Router
} 											from 'react-router-dom';										

const DATA = require('../../config.json');

configure( {adapter: new Adapter()} );

describe('<SidebarUser /> rendering', () => {
	let store, sidebaruser;
	let path = DATA.ims;
	const  options = {
		select : ['Name', 'DisplayName', 'JobTitle', 'Email', 'Skype'],
		query: 'TypeIs:User',
	};
	const mockStore = (createMockStore as any)();
	beforeEach( () => {
        store = mockStore({
			sensenet: {
				session: {
					user: {
                        userName: 'TestUser1',
                        fullName: 'Test User 1'
					}
				}
            },
            user: null
		});
        sidebaruser = mount(
            <Router>
                <Provider store={store}>
                    <SidebarUser /> 
                </Provider>
            </Router>
        ); 
    }); 
	
	// test Snapshot 
	it('Match to snapshot', () => {
        expect(toJson(sidebaruser)).toMatchSnapshot();
        sidebaruser.setState({
            usAvatar: 'not_empty'
        });
        expect(toJson(sidebaruser)).toMatchSnapshot();
    });
    
    it('Test props', () => {
        const sidebarChildren = sidebaruser.children().children().children().children();
        expect(sidebarChildren.props().fullName).toBe('Test User 1');
        expect(sidebarChildren.props().userName).toBe('TestUser1');
    });
});
