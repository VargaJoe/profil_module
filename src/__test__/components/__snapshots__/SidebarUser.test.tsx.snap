// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarUser /> rendering Match to snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(SidebarUser)>
        <SidebarUser
          dispatch={[Function]}
          fullName="Test User 1"
          user={null}
          userName="TestUser1"
        >
          <Link
            replace={false}
            to="/user/:TestUser1"
          >
            <a
              href="/user/:TestUser1"
              onClick={[Function]}
            >
              <div
                className="sn_sidebar__user"
              >
                <span
                  className="sn_sidebar__user__avatar--default"
                >
                  <i
                    className="fi flaticon-user"
                  />
                </span>
                <div
                  className="sn_sidebar__user__name"
                >
                  Test User 1
                </div>
              </div>
            </a>
          </Link>
        </SidebarUser>
      </Connect(SidebarUser)>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`<SidebarUser /> rendering Match to snapshot 2`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(SidebarUser)>
        <SidebarUser
          dispatch={[Function]}
          fullName="Test User 1"
          user={null}
          userName="TestUser1"
        >
          <Link
            replace={false}
            to="/user/:TestUser1"
          >
            <a
              href="/user/:TestUser1"
              onClick={[Function]}
            >
              <div
                className="sn_sidebar__user"
              >
                <span
                  className="sn_sidebar__user__avatar--default"
                >
                  <i
                    className="fi flaticon-user"
                  />
                </span>
                <div
                  className="sn_sidebar__user__name"
                >
                  Test User 1
                </div>
              </div>
            </a>
          </Link>
        </SidebarUser>
      </Connect(SidebarUser)>
    </Provider>
  </Router>
</BrowserRouter>
`;
